crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -66,10 +67,26 @@ fn top_level_statement(
         submodule(module_parser),
         module_declaration().then_ignore(force(just(Token::Semicolon))),
         use_statement().then_ignore(force(just(Token::Semicolon))),
+        global_declaration().then_ignore(force(just(Token::Semicolon))),
     ))
     .recover_via(top_level_statement_recovery())
 }
 
+fn global_declaration() -> impl NoirParser<TopLevelStatement> {
+    // let p = ignore_then_commit(
+    //     keyword(Keyword::Const).labelled("global const"),
+    //     ident().map(Pattern::Identifier),
+    // ); // TODO: this needs to be fixed, keep for now just to get stuff tested
+    let p = ignore_then_commit(
+        keyword(Keyword::Let).labelled("global const"),
+        ident().map(Pattern::Identifier),
+    );
+    let p = p.then(optional_type_annotation()); //TODO: rust requires annotation of global consts, perhaps we should as well and use a diff parser
```

Comment:
If we're using `then_commit` elsewhere, we should be consistent and use it here to.

For background, `then_commit` and friends are our versions of existing chumsky functions:
`then` -> `then_commit`,
`ignore_then` -> `ignore_then_commit`,
`then_ignore` -> `then_commit_ignore`

Compared to the normal versions the 'commit' version says "if the first thing parses I want to commit to parsing the second thing with no backtracking." This improves error messages when used right but can be subtle sometimes. For example, it is usually good to use after a keyword like `fn` where we know we want to commit to parsing a function since there isn't anything else that could come after such a keyword. It wouldn't be correct however to use it when parsing a tuple since it may be the parentheses were just a non-tuple parenthesized expression.
`